### DISCUSSION TITLE
Feature Request  :  Create Function without Brackets

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4087] [Ideas]
I know it may sounds silly, but for aesthetic reasons, can we please have the option to define a function without brackets, as the values are already hard coded, instead of FUNCTION Total_Quantity(), it will be FUNCTION Total_Quantity

(Disclosure : I used DAX and it is not required )

see example
```
SET default_collation='nocase';
CREATE TABLE lineitem(l_returnflag string, l_quantity numeric);
INSERT INTO lineitem VALUES ('A',150), ('b',300), ('c',200) ,('D',100),;
CREATE FUNCTION Total_Quantity() as (select sum (l_quantity ) from lineitem );
CREATE FUNCTION Quantity_A()  as (select sum(l_quantity) filter ( l_returnflag = 'a') from lineitem) ;

select   l_returnflag
        , sum(l_quantity)
        , Quantity_A() 
        , Total_Quantity() as Total
        , sum(l_quantity)/Total_Quantity() as Percentage 
        from lineitem group by all ;
```

**Comment by djouallah** [comment]
please ignore it, it is much more powerful than a measure !!!!

**Comment by lnkuiper** [comment]
We can change the parser to accept something like this:
```sql
create function test as (select 42);
```

However, when the function is called, we still need the parenthesis to figure out whether we are calling a function or selecting a column, e.g.:
```sql
select test;
-- Error: Binder Error: Referenced column "test" not found in FROM clause!
select test();
--┌────────┐
--│ test() │
--├────────┤
--│ 1      │
--└────────┘
```

Would that satisfy your request?

**Comment by djouallah** [comment]
@lnkuiper cheers don't worry actually it is better as it is , I thought of functions as a DAX measure, but it is much more !!!! will bother you later with Filter context and other shenanigans :)

**Comment by djouallah** [comment]
@lnkuiper hmm, it seems functions dont' accept table as a parameter ? 
CREATE FUNCTION sum_all(x,tbl) as (select sum (x ) from tbl );   does not Work ?

**Reply by lnkuiper** [reply]
No, that does not work. Functions defined by `CREATE FUNCTION ... AS ...` are like scalar functions, they cannot have table names as a parameter.

You will have to do something like:
```sql
create function sum_all(x) as sum(x);
select sum_all(x) from tbl;
```

**Reply by lnkuiper** [reply]
You can also create a table function by doing something like this:
```sql
CREATE FUNCTION sum_all(x) AS TABLE SELECT sum(x) FROM tbl;
```
